src/sciline/pipeline.py

```diff
@@ -422,12 +406,12 @@ def set_param_table(self, params: ParamTable) -> None:
             for index, label in zip(params.index, values):
                 self._set_provider(
                     Item((Label(tp=params.row_dim, index=index),), param_name),
-                    lambda label=label: label,
+                    _provider_for_constant(label),
```

Comment:
Note the logic in `ArgSpec.from_provider`, implements a hard requirement for not allowing arguments without type hints as per #115. Thus, the old lambda does not work any more.